File: Introduccion-POO/Ejercicio-1/Persona.py
class Persona:
    def __init__(self, nombre, ciudad, edad):
        self.nombre = nombre
        self.ciudad = ciudad
        self.edad = edad

    def saludo(self):
        print(f"Hola, soy {self.nombre} de {self.ciudad}")

    def es_mayor(self):
        if self.edad > 18:
            print(f"Sí es mayor y tiene {self.edad} años")
        else:
            print(f"No es mayor de edad y tiene {self.edad} años")


# Parte principal (equivalente a main en Java)
if __name__ == "__main__":
    p1 = Persona("David", "Bogota", 32)
    p2 = Persona("Roxana", "Buenos Aires", 15)
    p3 = Persona("Rai", "La Paz", 22)

    print("---------------Lista de Personas------------")
    p1.saludo()
    p2.saludo()
    p3.saludo()

    print("--------------¿Quién es mayor?----------------")
    p1.es_mayor()
    p2.es_mayor()
    p3.es_mayor()

File: Introduccion-POO/Ejercicio-1/Persona.java
public class Persona {
    private String nombre, cuidad;
    private int edad;

    public Persona(String nombre, String cuidad, int edad) {
        this.nombre = nombre;
        this.cuidad = cuidad;
        this.edad = edad;
    }
    
    public void Saludo()
    {
        System.out.println("Hola, soy "+nombre+" de "+cuidad);
    }
    
    public void Esmayor()
    {
        if(edad > 18)
        {
            System.out.println("Si es mayor y tiene "+edad+" anos");
        }
        else
            System.out.println("No es mayor de edad y tiene "+edad+" anos");
    }
    
    public static void main(String[] args) {
        // TODO code application logic here
        Persona p1=new Persona("David", "Bogota", 32);        
        Persona p2=new Persona("Roxana", "Buenos Aires", 15);
        Persona p3=new Persona("Rai", "La Paz", 22);
        
        System.out.println("---------------Lista de Personas------------");
        p1.Saludo();
        p2.Saludo();
        p3.Saludo();
        
        System.out.println("--------------Quien es mayor?----------------");
        p1.Esmayor();
        p2.Esmayor();
        p3.Esmayor();
    }
    
}